[Misc] Refactor: Improve argument handling for `conda` command

reidliu41 opened the pull request:
## Essential Elements of an Effective PR Description Checklist
- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.


## Purpose

Remove shell=True and add command blacklist to prevent command injection

The original code uses shell=True to support string commands, which may lead to command injection risks if combined with untrusted inputs or environment variables, e.g. `CONDA_EXE="conda; rm -rf /"`.

Changes
- Removed shell=True usage and enforced shell=False for all subprocess calls to prevent shell injection.
- Supported string commands by using shlex.split to safely split the command.
- Introduced a command blacklist to block the execution of dangerous system commands such as: rm, wget, curl, nc, bash, sh, mkfs, dd, scp, ftp, poweroff, reboot, init

## Test Plan

## Test Result

## (Optional) Documentation Update

<!--- pyml disable-next-line no-emphasis-as-heading -->

reidliu41 commented:
cc @DarkLight1337

DarkLight1337 reviewed (approved):
LGTM but I'll have @youkaichao @mgoin take another look just to be sure

youkaichao reviewed (commented):
why do we need to take care of security here? who can inject arbitrary command in this file? this file follows https://github.com/pytorch/pytorch/blob/main/torch/utils/collect_env.py .

reidliu41 committed:
Refactor: Improve argument handling for usage: conda [-h] [-v] [--no-plugins] [-V] COMMAND ...

conda is a tool for managing and deploying applications, environments and packages.

options:
  -h, --help            Show this help message and exit.
  -v, --verbose         Can be used multiple times. Once for detailed output,
                        twice for INFO logging, thrice for DEBUG logging, four
                        times for TRACE logging.
  --no-plugins          Disable all plugins that are not built into conda.
  -V, --version         Show the conda version number and exit.

commands:
  The following built-in and plugins subcommands are available.

  COMMAND
    activate            Activate a conda environment.
    clean               Remove unused packages and caches.
    commands            List all available conda subcommands (including those
                        from plugins). Generally only used by tab-completion.
    compare             Compare packages between conda environments.
    config              Modify configuration values in .condarc.
    content-trust       Signing and verification tools for Conda
    create              Create a new conda environment from a list of
                        specified packages.
    deactivate          Deactivate the current active conda environment.
    doctor              Display a health report for your environment.
    export              Export a given environment
    info                Display information about current conda install.
    init                Initialize conda for shell interaction.
    install             Install a list of packages into a specified conda
                        environment.
    list                List installed packages in a conda environment.
    notices             Retrieve latest channel notifications.
    package             Create low-level conda packages. (EXPERIMENTAL)
    remove (uninstall)  Remove a list of packages from a specified conda
                        environment.
    rename              Rename an existing environment.
    repoquery           Advanced search for repodata.
    run                 Run an executable in a conda environment.
    search              Search for packages and display associated information
                        using the MatchSpec format.
    tos                 A subcommand for viewing, accepting, rejecting, and
                        otherwise interacting with a channel's Terms of
                        Service (ToS). This plugin periodically checks for
                        updated Terms of Service for the active/selected
                        channels. Channels with a Terms of Service will need
                        to be accepted or rejected prior to use. Conda will
                        only allow package installation from channels without
                        a Terms of Service or with an accepted Terms of
                        Service. Attempting to use a channel with a rejected
                        Terms of Service will result in an error.
    update (upgrade)    Update conda packages to the latest compatible
                        version. command

Signed-off-by: reidliu41 <reid201711@gmail.com>

reidliu41 commented:
Thank you for your insightful feedback. It helped me refine my approach, and
  I've updated the PR with a more targeted solution.

  You've raised very valid points. After a deeper look, I agree this isn't a practical
  security vulnerability in the current context. My original goal was always aimed at
  proactive code hardening—what @russellb  rightly called "improving the
  code"—and I apologize if the initial "security" framing was a distraction.

  The new change is now surgically focused on the one case where an environment variable
  is used.
